File: mainlibrary/src/main/java/com/aite/mainlibrary/activity/allsetting/editSosUser/RetrofitInterface.java
package com.aite.mainlibrary.activity.allsetting.editSosUser;

import com.aite.mainlibrary.Mainbean.AddbinduserfamilyBean;
import com.aite.mainlibrary.Mainbean.BinduserfamilyTypeBean;
import com.aite.mainlibrary.Mainbean.MoneyBookBean;
import com.aite.mainlibrary.Mainbean.StateCodeBean;
import com.aite.mainlibrary.Mainbean.SuccessCodeBean;
import com.aite.mainlibrary.activity.allstep.vipStep.StepInformationBean;
import com.lzy.basemodule.bean.BaseData;
import com.lzy.basemodule.bean.BaseData2;
import com.lzy.basemodule.bean.BaseDataX;

import java.io.File;
import java.util.List;
import java.util.Map;

import io.reactivex.Flowable;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.PartMap;
import retrofit2.http.Query;

/**
 * @Auther: liziyang
 * @datetime: 2020-01-11
 * @desc:
 */
public interface RetrofitInterface {
    /**
     * 得到紧急联系人信息
     *
     * @param key
     * @param id  紧急联系人id
     * @return
     */
    @GET("index.php?act=member_contact&op=my_detail")
    Flowable<BaseData<SosUserInformationBean>> onGetSosInformationData(
            @Query("key") String key,
            @Query("id") String id
    );

    /**
     * 得到紧急联系人关系信息
     *
     * @param key
     * @return
     */
    @GET("index.php?act=member_associate&op=relation")
    Flowable<ResponseBody> onGetSosRelationshipData(
            @Query("key") String key
    );

    /**
     * key	post	字符串	必须			会员登录key
     * id	post	整型	可选			紧急联系人id
     * type	post	整型	必须			保存类型 1添加 2修改(需上传紧急联系人id)
     * avatar	post	文件	必须	1		头像文件参数
     * realname	post	字符串	必须			真实姓名
     * mobile	post	字符串	必须			手机号
     * relation	post	整型	必须			关系id
     * is_default	post	整型	可选			是否 默认1- 否0
     *
     * @PartMap Map<String, ResponseBody> responseBodyMap
     * @Part("key") String key,
     * @Part("id") String id,
     * @Part("type") String type,
     * @Part("avatar") MultipartBody.Part avatar,
     * @Part("realname") String realname,
     * @Part("mobile") String mobile,
     * @Part("relation") String relation,
     * @Part("is_default") String is_default
     */

    @Multipart
//    @FormUrlEncoded
    @POST("index.php?act=member_contact&op=my_save")
    Flowable<BaseData<StateCodeBean>> onSaveEditSosUserInformation(
            @Part List<MultipartBody.Part> content
    );

    /**
     * APP签到页
     *
     * @param key
     * @return
     */
    @GET("index.php?act=member_sign&op=appindex")
    Flowable<BaseData<StepInformationBean>> onGetUserStepData(
            @Query("key") String key
    );

    /**
     * APP签到 点击签到
     *
     * @param key
     * @return
     */
    @GET("index.php?act=member_sign&op=sign")
    Flowable<ResponseBody> onUserStep(
            @Query("key") String key
    );

    /**
     * APP-安卓版本号 判断是否自动更新
     *
     * @param key
     * @return
     */
    @GET("index.php?act=index&op=android_version")
    Flowable<ResponseBody> onGetUpdateVersion(
            @Query("key") String key
    );

    /**
     * APP账单明细 获取会员账单明细
     *
     * @param key
     * @param curpage
     * @return
     */
    @GET("index.php?act=predeposit&op=yck_list")
    Flowable<BaseDataX<MoneyBookBean>> onBillingDetails(
            @Query("key") String key,
            @Query("curpage") int curpage
    );
}

File: mainlibrary/src/main/java/com/aite/mainlibrary/activity/allsetting/feedback/RetrofitInterface.kt
package com.aite.mainlibrary.activity.allsetting.feedback

import com.aite.mainlibrary.activity.allsetting.editSosUser.SosUserInformationBean
import com.lzy.basemodule.bean.BaseData
import io.reactivex.Flowable
import okhttp3.ResponseBody
import retrofit2.http.*

/**

 * @Auther: liziyang

 * @datetime: 2020-01-19

 * @desc:

 */
interface RetrofitInterface {
    @GET("index.php?act=member_feedback&op=feedback_typelist")
    fun onGetFeedTypeList(
            @Query("key") key: String
    ): Flowable<ResponseBody>

    /**
     * 会员添加意见反馈
     *
    参数名字	提交方式	类型	是否必须	默认值	其他	说明	test
    key	post	字符串	必须			会员登录key
    049507bc7fdedc052495a82d9c63d7d7
    feedback	post	字符串	必须			反馈内容
    dsaf
    type_id	post	整型	必须			反馈类型编号
    1
    client_type	post	整型	必须	1		反馈客户端 1来自手机端2来自PC端
     */
    @POST("index.php?act=member_feedback&op=feedback_add")
    @FormUrlEncoded
    fun onCommitFeed(
            @Field("key") key: String,
            @Field("feedback") feedback: String,
            @Field("type_id") type_id: String,
            @Field("client_type") client_type: String

    ): Flowable<ResponseBody>
}